app/scripts/modules/core/src/pipeline/config/triggers/artifacts/helm/HelmArtifactEditor.tsx

```diff
@@ -4,71 +4,98 @@ import { Option } from 'react-select';
 import { ArtifactTypePatterns } from 'core/artifact';
 import { IArtifact, IArtifactEditorProps, IArtifactKindConfig } from 'core/domain';
 import { StageConfigField } from 'core/pipeline';
-import { TetheredSelect } from 'core/presentation';
-
+import { TetheredSelect, TetheredCreatable } from 'core/presentation';
 import { ArtifactService } from '../ArtifactService';
 import { cloneDeep } from 'lodash';
+import { Spinner } from 'core/widgets';
 
 const TYPE = 'helm/chart';
+// taken from https://github.com/semver/semver/issues/232
+const SEMVER = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$/;
 
 interface IHelmArtifactEditorState {
   names: string[];
-  versions: string[];
+  versions: Option<string>[];
+  versionsLoading: boolean;
+  namesLoading: boolean;
 }
 
 class HelmEditor extends React.Component<IArtifactEditorProps, IHelmArtifactEditorState> {
+  public state: IHelmArtifactEditorState = {
+    names: [],
+    versions: [],
+    versionsLoading: true,
+    namesLoading: true,
+  };
+
   constructor(props: IArtifactEditorProps) {
     super(props);
-    if (props.artifact.type !== TYPE) {
+    const { artifact } = this.props;
```

Comment:
If we're going to destructure this, let's go ahead and replace `props.artifact` with `artifact` below on line 35